flaml/automl/model.py

```diff
@@ -1486,8 +1493,12 @@ def cost_relative2lgbm(cls):
         return 1.9
 
     def __init__(self, task="binary", **params):
+        if isinstance(task, str):
+            from flaml.automl.task.factory import task_factory
+
+            task = task_factory(task)
         super().__init__(task, **params)
-        if "regression" in task:
+        if task.is_regression():

```

Comment:
There are more issues like this: 1453, 1470, 1484. There should be more.

Reply:
why is this an issue?

Reply:
`task in [list]` is actually perfectly legitimate if `task` is a `Task` object, as the latter implements a comparison operator to strings. There are so many of these scattered all over the code that we felt it was easier to support them as they are for now, and gradually clean them up later.

Reply:
On the other hand, `"regression" in task` would not work with that approach; but I did a full-text search of the code for `in task` and found no hits